### PR TITLE
Add Tooltips and Element ID's

### PR DESCRIPTION
**Pull request by Clonephaze**
I added tooltips to each relevant location in webui, to help explain each feature without needing to open the github page. I did not know what to put in the tooltip for stride, or the options in img2img so I left those blank for you to fill out if you're interested.

I also added element id's so that extensions like Config-Presets can latch on and control each configurable option.

### CODE REVIEW

**Comment by continue-revolution** [comment]
You may want to specify txt2img or img2img. Otherwize element ids for some items will not be unique.

**Comment by Clonephaze** [comment]
> You may want to specify txt2img or img2img. Otherwize element ids for some items will not be unique.

I'd like to but the way the extensions are set up they share most of the elements codes. The umiAI extension shares it's complete menu for some reason. Without knowing more about how that works I can't separate the two windows. The animated diff extension shares all but 4 of its menu elements.

**Comment by Clonephaze** [comment]
Ill have to see what I can do. As it stands they share the same menu so far as I can tell so I'll have to figure out how to separate them before I can do that. Today's my first day working with gradio so I've gotta study up a bit.

**Comment by continue-revolution** [comment]
Take your time. I'll convert it to draft. When you think you are ready, please feel free to convert it to ready for review.

**Comment by Clonephaze** [comment]
@continue-revolution That was actually fairly simple once I grasped how it separated the two tab's info.

File: scripts/animatediff_ui.py [hunk 1]
```diff
@@ -98,6 +98,7 @@ def __init__(self):
     def render(self, is_img2img: bool, model_dir: str):
         if not os.path.isdir(model_dir):
             os.mkdir(model_dir)
+        element_id_prefix = "ad-img2img-" if is_img2img else "ad-"
```
Comment (continue-revolution) [inline scripts/animatediff_ui.py:101]: better to have this format: "txt2img-ad-" or "img2img-ad-" as this is what A1111 has. You can also make this name shorter such as `elemid_prefix`

File: scripts/animatediff_ui.py [hunk 2]
```diff
@@ -120,34 +121,42 @@ def refresh_models(*inputs):
                     value=(self.params.model if self.params.model in model_list else None),
                     label="Motion module",
                     type="value",
+                    tooltip="Choose which motion module will be injected into the generation process.",
+                    elem_id=element_id_prefix + "motion-module",
```
Comment (continue-revolution) [inline scripts/animatediff_ui.py:125]: please use fstring like f"{element_id_prefix}motion-module"

File: scripts/animatediff_ui.py [hunk 3]
```diff
@@ -120,34 +121,42 @@ def refresh_models(*inputs):
...
                     minimum=0,
                     value=self.params.video_length,
                     label="Number of frames",
                     precision=0,
+                    tooltip="Total length of video in frames.",
+                    elem_id=element_id_prefix + "video-length",
                 )
                 self.params.fps = gr.Number(
-                    value=self.params.fps, label="FPS", precision=0
+                    value=self.params.fps, label="FPS", precision=0, tooltip="How many frames per second the gif will run.", elem_id=element_id_prefix + "fps"
```
Comment (continue-revolution) [inline scripts/animatediff_ui.py:144]: start a new line if too long

File: scripts/animatediff_ui.py [hunk 4]
```diff
@@ -156,40 +165,51 @@ def refresh_models(*inputs):
...
                 self.params.interp = gr.Radio(
                     choices=["Off", "FILM"],
                     label="Frame Interpolation",
+                    tooltip="Interpolate between frames with Deforum's FILM implementation. Requires Deforum extension.",
+                    elem_id=element_id_prefix + "interp-choice",
                     value=self.params.interp
                 )
                 self.params.interp_x = gr.Number(
-                    value=self.params.interp_x, label="Interp X", precision=0
+                    value=self.params.interp_x, label="Interp X", precision=0, tooltip="Replace each input frame with X interpolated output frames.", elem_id=element_id_prefix + "interp-x"
```
Comment (continue-revolution) [inline scripts/animatediff_ui.py:212]: start a new line if too long

**Pull request merged**